api/v1beta1/cluster_types.go

```diff
@@ -517,6 +518,15 @@ type Topology struct {
 	// The name of the ClusterClass object to create the topology.
 	Class string `json:"class"`
 
+	// The namespace of the ClusterClass object to create the topology. Empty namespace assumes the namespace of the cluster object.
+	// Class namespace changes are not supported by the rebase procedure, as different CC namespace uses namespace-local templates.
+	// Cluster templates namespace modification is not allowed.
+	// +optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:Pattern="^[a-z0-9](?:[-a-z0-9]*[a-z0-9])?(?:\\.[a-z0-9](?:[-a-z0-9]*[a-z0-9])?)*$"
+	ClassNamespace string `json:"classNamespace,omitempty"`
```

Comment:
do we have consensus about adding this field to the API?

i mean, it seems necessary for this change, but i don't recall what we might have agreed upon in the office hours.

Reply:
It’s actually a superset of #11352, I believe we have a consensus on the actual PR in the review https://github.com/kubernetes-sigs/cluster-api/pull/11352#discussion_r1856631550 (correct me if I’m wrong)